src/_pytest/config/__init__.py

```diff
@@ -899,6 +900,20 @@ class InvocationParams:
         dir: Path
         """The directory from which :func:`pytest.main` was invoked."""
 
+    @final
```

Comment:
Since we don't expose this via the public API (contrary to [e.g. `Config` above](https://github.com/pytest-dev/pytest/blob/eb8b3ad929cdbaf996be1bb16b37abd7762cceeb/src/pytest/__init__.py#L10)), I don't think this is needed.

Reply:
Got it. I was following the example set by [`Config.IncovationParams`](https://github.com/pytest-dev/pytest/blob/eb8b3ad929cdbaf996be1bb16b37abd7762cceeb/src/_pytest/config/__init__.py#L878), which does use `@final`.